lib/logidze/versioned_association.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+require 'active_support'
+
+module Logidze
+  module VersionedAssociation
+    module SingularAssociation
+      def reader(force_reload = false)
```

Comment:
And how to handle association loading? I.e. `post.at(...).user`?
How to prevent double-`at!`?

I have another idea: we can use [`stale_target?`](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/associations/association.rb#L75) to handle reload. Just:

```ruby
def stale_target?
  logidze_state? || super
end

def logidze_stale?
  # we can store logidze_requested_ts in the association too
  loaded? && !inversed? && (owner.logidze_requested_ts != logidze_requested_ts)
end
```

That should work for all kind of associations.

But I found one more problem with collection associations: we have to think about `empty?`, `any?`, `blank?` methods (which use SQL if target has not been loaded) and also `update_all`, `delete_all`.
For presence-like checks we can force load target; and I think we should raise an error when a user tries to use `update_all` / `delete_all` on the past version.
Potential caveat:

```ruby
post = Post.find(...)
post.comments.size #=> 2

# rollback and persist
post.undo!

# no raise
post.comments.update_all(...)
```

Too complicated. We need more specs, first of all)

UPD: one more issue: `post.comment_ids`.

Maybe it would be easier to write our own `CollectionAssociation` that mimics the AR one and make it readonly, than to monkey-patch everything.